### ISSUE TITLE
Warnings regression in coreclr build

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#24448]
These warnings are newly seen in the coreclr product build (specifically, System.Private.Corelib.dll build):

```
BUILD: Commencing build of System.Private.CoreLib for Windows_NT.x64.Checked
Tools are already initialized.
Running: dotnet restore /nologo /verbosity:minimal /clp:Summary /nodeReuse:false          /l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log          /p:RestoreDefaultOptimizationDataPackage=false /p:PortableBuild=true          /p:UsePartialNGENOptimization=false /maxcpucount /p:IncludeRestoreOnlyProjects=true /p:ArcadeBuild=true          F:\workspace\_work\1\s\src\build.proj          /flp:Verbosity=normal;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.log" /flp1:WarningsOnly;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.wrn" /flp2:ErrorsOnly;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.err" /p:__BuildOS=Windows_NT /p:__BuildType=Checked /p:__BuildArch=x64 /p:RestoreDuringBuild=true   /p:BuildManagedTools=true /p:BuildNugetPackage=true
Restoring packages for C:\Users\vsagent\AppData\Local\Temp\06643f3e-3b83-46b3-9bbf-78d881a2cfc5...
Installing Microsoft.Build.Traversal 1.0.52.
Restoring packages for C:\Users\vsagent\AppData\Local\Temp\06643f3e-3b83-46b3-9bbf-78d881a2cfc5...
Installing Microsoft.DotNet.Arcade.Sdk 1.0.0-beta.19229.8.
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\Directory.Build.props(4,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\dir.props (5,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\.nuget\init\init.csproj]
C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props(17,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\dependencies.props (14,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\.nuget\init\init.csproj]
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj]
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj]
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
```

**Comment by BruceForstall** [comment]
@AaronRobinsonMSFT

**Comment by AaronRobinsonMSFT** [comment]
@RussKeldorph this issue should be addressed when the last of the BuildTools dependencies are removed.

**Comment by RussKeldorph** [comment]
@AaronRobinsonMSFT Can this be addressed now, or do we need to wait until the test dependencies are fixed as well?

@jashook @elinor-fung

**Comment by AaronRobinsonMSFT** [comment]
I believe @elinor-fung fixed these warnings.

**Comment by RussKeldorph** [comment]
@AaronRobinsonMSFT This is what I see from a build this morning https://dev.azure.com/dnceng/public/_build/results?buildId=225141.  The warnings look similar, so I assumed this still needs work.

```
BUILD: Commencing build of System.Private.CoreLib for Windows_NT.x64.Checked
Installing dotnet using Arcade...
running: powershell -NoProfile -ExecutionPolicy unrestricted -Command ". F:\workspace\_work\1\s\eng\configure-toolset.ps1; . F:\workspace\_work\1\s\eng\common\tools.ps1; InitializeBuildTool"

Name                           Value                                                                                   
----                           -----                                                                                   
Path                           F:\workspace\_work\1\s\.dotnet\dotnet.exe                                               
Command                        msbuild                                                                                 
Tool                           dotnet                                                                                  
Framework                      netcoreapp2.1                                                                           


Running: dotnet restore /nologo /verbosity:minimal /clp:Summary /nodeReuse:false          /p:PortableBuild=true /maxcpucount /p:IncludeRestoreOnlyProjects=true /p:ArcadeBuild=true          F:\workspace\_work\1\s\src\build.proj          /flp:Verbosity=normal;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.log" /flp1:WarningsOnly;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.wrn" /flp2:ErrorsOnly;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.err" /p:__BuildOS=Windows_NT /p:__BuildType=Checked /p:__BuildArch=x64 /p:RestoreDuringBuild=true   /p:BuildManagedTools=true     /p:ContinuousIntegrationBuild=true
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\runincontext\runincontext.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\runincontext\runincontext.csproj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
  Restore completed in 62.09 ms for F:\workspace\_work\1\s\src\build.proj.
  Restore completed in 566.04 ms for F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj.
  Restore completed in 446.44 ms for F:\workspace\_work\1\s\src\tools\runincontext\runincontext.csproj.
F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj : warning NU1604: Project dependency System.Reflection.Metadata does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj : warning NU1701: Package 'System.Collections.Immutable 1.1.36' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj : warning NU1701: Package 'System.Reflection.Metadata 1.0.21' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project. [F:\workspace\_work\1\s\src\build.proj]
  Restore completed in 2.01 sec for F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj.
  Restore completed in 3.31 sec for F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj.
```

**Comment by elinor-fung** [comment]
I handled the duplicate imports for packaging, signing, and optdata (so the one from init.csproj is gone), but did not get around to going back to try to fix the ones from System.Private.CoreLib and managed tools.

I think I could easily avoid those warnings right now by just not adding those imports if `ArcadeBuild` is true. A nicer fix should also be possible before tests are moved off of BuildTools, but it may be a bit more complicated / require extra work (compared to if the test dependencies on BuildTools were removed).
